### PR TITLE
Upsert Entity Record documents, fix RunView not paginating results 

### PR DESCRIPTION
**Pull request by JS-BC**

### CODE REVIEW

File: packages/AI/Providers/Vectors-Pinecone/src/models/PineconeDatabase.ts [hunk 1]
```diff
@@ -186,17 +186,19 @@ export class PineconeDatabase extends VectorDBBase {
             response.Duplicates = [];
             for(const match of queryResponse.data.matches){
                 const record: {id: string, score: number, metadata: any} = match;
-                if(!record.metadata || !record.metadata.PrimaryKeys){
+                if(!record || !record.id){
```
Comment (AN-BC) [inline packages/AI/Providers/Vectors-Pinecone/src/models/PineconeDatabase.ts:189]: @js-bc - is this something from Pinecone, hence the harcoded record.id?

File: packages/AI/Providers/Vectors-Pinecone/src/models/PineconeDatabase.ts [hunk 2]
```diff
@@ -186,17 +186,19 @@ export class PineconeDatabase extends VectorDBBase {
...
+                const splitID: string[] = record.id.split('_');
+                if(splitID.length !== 3){
+                    LogError(`Invalid vector record ID: ${record.id}`);
                     continue;
                 }
 
                 let duplicate: PotentialDuplicate = new PotentialDuplicate();
-                duplicate.LoadFromList(metadata.PrimaryKeys);
+                duplicate.KeyValuePairs.push({ FieldName: 'ID', Value: splitID[2] });
                 duplicate.ProbabilityScore = record.score;
```
Comment (AN-BC) [inline packages/AI/Providers/Vectors-Pinecone/src/models/PineconeDatabase.ts:202]: @js-bc I'm confused why there is code related to duplicate data in the PineconeDatabase class - that class is specific to a given provider, what does this have to do with the generic idea of duplicate detection?

File: packages/AI/Vectors/Core/src/models/VectorBase.ts [hunk 3]
```diff
@@ -46,21 +46,19 @@ export class VectorBase {
...
-          MaxRows: pageSize,
-          OffsetRows: Math.max(0, (pageNumber - 1) * pageSize),
-        };
-    
-        const rvResult: RunViewResult<T> = await this._runView.RunView<T>(params, this.CurrentUser);
+        const rvResult: RunViewResult<T> = await this._runView.RunView<T>({
+            EntityName: entity.Name,
+            ResultType: 'entity_object' as const,
+            MaxRows: pageSize,
+            OffsetRows: Math.max(0, (pageNumber - 1) * pageSize),
```
Comment (AN-BC) [inline packages/AI/Vectors/Core/src/models/VectorBase.ts:53]: @js-bc - this is fine for now, but for this PR change property to "StartRow" and in the future, the code in SQLServerDataProvider must be changed to use the sort order of the view as defined by the stored view and/or the RunViewParams. The proper way to do this is with a CTE that encapsulates the query and uses numbered rows in SQL Server (get ChatGPT to help) and then from there just picks the specific rows we want. You can DEFER doing the proper SQLServer approach for just now to get 2.0 out the door but we need to change this very soon after the 2.0 release

**Review by AN-BC — COMMENTED** [review]
Comments in the code in various places @js-bc

File: packages/AI/Providers/Vectors-Pinecone/src/models/PineconeDatabase.ts (hunk 1 above)
Comment (JS-BC) [reply packages/AI/Providers/Vectors-Pinecone/src/models/PineconeDatabase.ts:189]: Yes

File: packages/AI/Providers/Vectors-Pinecone/src/models/PineconeDatabase.ts (hunk 2 above)
Comment (JS-BC) [reply packages/AI/Providers/Vectors-Pinecone/src/models/PineconeDatabase.ts]: You're right. I've moved the function out of the VectorDBBase class and put it inside the DuplicateRecordDetector class
